purchase_sale_stock_inter_company/tests/test_inter_company_purchase_sale_stock.py

```diff
@@ -0,0 +1,440 @@
+            len(so_lots),
+            len(po_lots),
+            msg="There aren't the same number of lots on both moves",
+        )
+        self.assertNotEqual(
+            so_lots, po_lots, msg="The lots of the moves should be different objects"
+        )
+        self.assertEqual(
+            so_lots.mapped("name"),
+            po_lots.mapped("name"),
+            msg="The lots should have the same name in both moves",
+        )
+
+    def test_sync_picking_lot_and_qty_with_move_diff_for_multiple_products(self):
+        """
+        Test that the lot is synchronized on the moves
+        by searching or creating a new lot in the company of destination
+        """
+        po_form = Form(self.env["purchase.order"])
+        po_form.company_id = self.company_a
+        po_form.partner_id = self.partner_company_b
+
+        with po_form.order_line.new() as line_form:
+            line_form.product_id = self.stockable_product_serial
+            line_form.product_qty = 1
+            line_form.price_unit = 450.0
+        with po_form.order_line.new() as line_form:
+            line_form.product_id = self.stockable_product
+            line_form.product_qty = 50
+            line_form.price_unit = 1000.0
+        self.purchase_company_a = po_form.save()
+
+        purchase = self.purchase_company_a
+        sale = self._approve_po()
+
+        # validate the SO picking
+        po_picking_id = purchase.picking_ids
+        so_picking_id = sale.picking_ids
+        so_move = so_picking_id.move_ids
+        so_move.move_line_ids = [
+            (
+                0,
+                0,
+                {
+                    "location_id": so_move.location_id.id,
+                    "location_dest_id": so_move.location_dest_id.id,
+                    "product_id": self.stockable_product_serial.id,
+                    "quantity": 1,
+                    "lot_id": self.serial_1.id,
+                    "picking_id": so_picking_id.id,
+                },
+            ),
+            (
+                0,
+                0,
+                {
+                    "location_id": so_move.location_id.id,
+                    "location_dest_id": so_move.location_dest_id.id,
+                    "product_id": self.stockable_product_serial.id,
+                    "quantity": 1,
+                    "lot_id": self.serial_2.id,
+                    "picking_id": so_picking_id.id,
+                },
+            ),
+            (
+                0,
+                0,
+                {
+                    "location_id": so_move.location_id.id,
+                    "location_dest_id": so_move.location_dest_id.id,
+                    "product_id": self.stockable_product.id,
+                    "quantity": 30,
+                    "lot_id": False,  # No lot for stockable product
+                    "picking_id": so_picking_id.id,
+                },
+            ),
+        ]
+        so_picking_id.sudo().button_validate()
```

Comment:
Test is failing here, because we try to deliver only 30 of the 50 requested "stockable_product" and button_validate is opening the dialog to create a backorder...

```suggestion
        res_dict = so_picking_id.sudo().button_validate()
        if isinstance(res_dict, dict) and "context" in res_dict:
            self.env["stock.backorder.confirmation"].with_context(
                **res_dict["context"]
            ).process()
```
I think using the same code as in previous tests with backorder creation would make this test pass.